frontend-ts/src/components/Record.tsx

```diff
@@ -16,7 +18,14 @@ export const Record: React.FC<RecordProps> = ({ capturing, setCapturing, recordi
   const [recordedChunks, setRecordedChunks] = useState<Blob[]>([]);
   const [mediaStream, setMediaStream] = useState<MediaStream | null>(null);
   const [timer, setTimer] = useState(0);
+  const [sessionId, setSessionId] = useState<string>("");
 
+  useEffect(() => {
+      if(capturing) {
+          setSessionId(uuidv4()); // Set a new unique ID every time a recording starts
+      }
+  }, [capturing]);
```

Comment:
I should check once more, but I think I may [not need an effect here](https://react.dev/learn/you-might-not-need-an-effect)